### PR TITLE
feat: add check to test final url when redirect present

### PR DESCRIPTION
**Pull request by moshetanzer**
Hi @danielroe,

Intrested to get some good criticism and feedback.

To me biggest issue is that we shouldn't be checking domain redirect twice 🙈. (We can't move to front end because of cors, if was a way to implement in middleware would run twice I think 🤔..) and making it into a function. 

Also think we might need to update cache to be more efficient (in case where there is redirect).

Also, couldn't get it to work with $fetch, doesn't seem like it returns redirect like fetch - should I open an issue on ofetch - or is this on purpose. Could have done a manual if..else to check if final url was equal to inputted, though this was a better option. 

Not sure if you want to put the redirect message at the bottom with other info (else we need to move it closer to input field, didnt want to start adding slots in domain form or negative margins).

resolves https://github.com/danielroe/page-speed.dev/issues/9
Thanks for your work! 

## Summary by CodeRabbit

- **New Features**
	- Improved domain validation and navigation based on different environments for a better user experience.
	- Enhanced the template structure and text content related to Lighthouse APIs for Core Web Vitals and PageSpeed Insights.

- **Improvements**
	- Refactored domain validation logic in API endpoints to enhance code readability and maintainability.
	- Centralized domain validation logic to ensure proper domain validation before processing API calls.

### CODE REVIEW

**Comment by moshetanzer** [comment]
@danielroe any thoughts on this?

File: app.vue [hunk 1]
```diff
@@ -135,6 +135,9 @@ else {
         <TheDomainForm v-model:editing="editing" :domain="domain" />
       </div>
       <template v-if="!editing && domain">
+        <span v-if="(crux && crux.redirected) || (lighthouse && lighthouse.redirected)" class="text-gray-400">
+            Showing results for {{ crux?.domain }} {{ lighthouse?.domain }}
+          </span>
```
Comment (danielroe) [inline app.vue:140]: I think I'd be very happy not to display this additional line here but just display the actual domain shown in the domain form above.

File: server/api/crux/[domain].get.ts [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+      redirected = true
+      domain = response.url
+    }
+    else {
+      domain = `https://${domain}`
+    }
+  }
+  catch (error) {
+    throw createError({ message: 'Invalid domain', statusCode: 422 })
+  }
```
[the remark this answers is not in the document]
Comment (moshetanzer) [reply server/api/crux/[domain].get.ts:22]: Yeah, issue is unless we make another endpoint if i am not mistaken there are two requests coming from client? What would be your ideal way to do it, would you be happy with a call to check if there is a redirect and the if redirect just a client side redirect which will auto check correct url then? - happy to implement.

File: app.vue (hunk 1 above)
Comment (moshetanzer) [reply app.vue:140]: one of reasons did it like this was to keep interaction as simple as possible, this is alsoohow pagespeed had done it, but i hear you. let me know..

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
Thank you so much! :heart:

I made a few changes but maybe you have some ideas to improve further 🙏

**Pull request merged**